frontend/src/metabase/query_builder/hooks/use-query-aggregations.ts

```diff
@@ -0,0 +1,47 @@
+import { useMemo } from "react";
+
+import * as Lib from "metabase-lib";
+
+import type { UpdateQueryHookProps } from "./types";
+
+export const useQueryAggregations = ({
+  query,
+  onQueryChange,
+  stageIndex,
+}: UpdateQueryHookProps) => {
+  const aggregations = useMemo(
+    () => Lib.aggregations(query, stageIndex),
+    [query, stageIndex],
+  );
+
+  const items = useMemo(
+    () =>
+      aggregations.map((aggregation, aggregationIndex) => {
+        const { displayName } = Lib.displayInfo(query, stageIndex, aggregation);
+
+        const operators = Lib.selectedAggregationOperators(
+          Lib.availableAggregationOperators(query, stageIndex),
+          aggregation,
+        );
+
+        const handleRemove = () => {
+          const nextQuery = Lib.removeClause(query, stageIndex, aggregation);
+          onQueryChange(nextQuery);
+        };
+
+        return {
+          aggregation,
+          aggregationIndex,
+          operators,
+          displayName,
+          handleRemove,
+        };
+      }),
+    [aggregations, onQueryChange, query, stageIndex],
+  );
+
+  return {
+    items,
+    onQueryChange,
```

Comment:
1. This hook gets `onQueryChange` as an argument and just returns it - redundant, let's remove it from returned value.
2. This does not need to be a hook. We can simplify things by extracting `getAggregationItems(query, stageIndex)` function and removing `handleRemove` from here, like I suggested earlier.

Consumers would decide if this needs to be "hooked".

```tsx
const aggregationItems = useMemo(() => getAggregationItems(query, stageIndex), [query, stageIndex]);

const handleRemove = (aggregation: Lib.AggregationClause) => {
  const nextQuery = Lib.removeClause(query, stageIndex, aggregation);
  onQueryChange(nextQuery);
};

return aggregationItems.map((item, index) => {
  return <div key={index} onClick={() => handleRemove(item.aggregation)} />
});
```

Reply:
i'll do that then

Reply:
updated. Let me know what you think

Reply:
Looks good :+1: